### ISSUE TITLE
runtime/explicit doesn't appear with particular class ctor

### ISSUE THREAD

**Issue by hrkn** [google/styleguide#178]
Hi,
I found cpplint doesn't warn [runtime/explicit] for some particular template classes.

Issue will happen if:
- Target class is template
- and the template argument includes close paren '`)`'
- and it has one or more non-explicit unary constructor

Factor of issue is [matching on tokens after class identifier](https://github.com/google/styleguide/blob/gh-pages/cpplint/cpplint.py#L2587).
At that match, close paren `)` was assigned to `token`, and [instance of `_ClassInfo` in `self.stack` was popped](https://github.com/google/styleguide/blob/gh-pages/cpplint/cpplint.py#L2615).

Here is a code to reproduce the issue.

```
class BareClass {
  // properly [runtime/explicit] warned
  BareClass(int i) {
  }
};

template <typename F>
class Function;

template <typename R, typename... Args>
class Function<R(Args...)> {
  // [runtime/explicit] doesn't appear
  template <typename F>
  Function(F fn) {
    // ...
  }
};
```

**Comment by vapier** [comment]
public cpplint development in this repository has been halted (see https://github.com/google/styleguide/pull/837). please migrate to https://github.com/cpplint/cpplint now.
